core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -324,7 +331,7 @@ object TopicCommand extends Logging {
 
     override def deleteTopic(opts: TopicCommandOptions): Unit = {
       val topics = getTopics(opts.topic, opts.excludeInternalTopics)
-      ensureTopicExists(topics, opts.topic)
+      ensureTopicExists(topics, opts.topic, !opts.ifExists)
```

Comment:
Same issue here as above.  `Admin#deleteTopics` will fail if the topic doesn't exist, regardless of line 334.

Reply:
Like we discussed.. getTopics will guard the issue you point out.. avoiding `getTopics` is again tricky due to regex/wildcard matching